src/AppInstallerCommonCore/Manifest/MsixManifestValidation.cpp

```diff
@@ -0,0 +1,159 @@
+        try
+        {
+            if (!minOSVersion.empty())
+            {
+                return std::make_optional<Msix::OSVersion>(minOSVersion);
+            }
+        }
+        catch (const std::exception&)
+        {
+            errors.emplace_back(ManifestError::InvalidFieldValue, "MinimumOSVersion", minOSVersion);
+        }
+        return std::nullopt;
+    }
+
+    void MsixManifestValidation::ValidateMsixManifestPackageFamilyName(
+        Utility::NormalizedString msixPackageFamilyName,
+        std::string manifestPackageFamilyName,
+        std::vector<ValidationError> &errors)
+    {
+        if (!manifestPackageFamilyName.empty())
+        {
+            if (manifestPackageFamilyName != msixPackageFamilyName)
+            {
+                errors.emplace_back(ManifestError::InstallerMsixInconsistencies, "PackageFamilyName", msixPackageFamilyName);
+            }
+        }
+        else
+        {
+            errors.emplace_back(ManifestError::OptionalFieldMissing, "PackageFamilyName", msixPackageFamilyName, m_validationErrorLevel);
+        }
+    }
+
+    void MsixManifestValidation::ValidateMsixManifestPackageVersion(
+        const Msix::PackageVersion &msixPackageVersion,
+        const Msix::PackageVersion &manifestPackageVersion,
+        std::vector<ValidationError> &errors)
+    {
+        if (msixPackageVersion != manifestPackageVersion)
+        {
+            errors.emplace_back(ManifestError::InstallerMsixInconsistencies, "PackageVersion", msixPackageVersion.ToString());
+        }
+    }
+
+    void MsixManifestValidation::ValidateMsixManifestMinOSVersion(
+        const std::optional<Msix::OSVersion>& msixMinOSVersion,
+        const std::optional<Msix::OSVersion>& manifestMinOSVersion,
+        std::string installerUrl,
+        std::vector<ValidationError> &errors)
+    {
+        if (!msixMinOSVersion.has_value())
+        {
+            errors.emplace_back(ManifestError::NoSupportedPlatforms, "InstallerUrl", installerUrl);
+        }
+        else if (manifestMinOSVersion.has_value())
+        {
+            if (msixMinOSVersion.value() != manifestMinOSVersion.value())
+            {
+                errors.emplace_back(ManifestError::InstallerMsixInconsistencies, "MinimumOSVersion", msixMinOSVersion.value().ToString());
+            }
+        }
+        else
+        {
+            errors.emplace_back(
+                ManifestError::OptionalFieldMissing,
+                "MinimumOSVersion",
+                msixMinOSVersion.value().ToString(),
+                m_validationErrorLevel);
+        }
+    }
+
+    void MsixManifestValidation::ValidateMsixManifestSignatureHash(
+        const std::shared_ptr<Msix::MsixInfo> msixInfo,
+        const Utility::SHA256::HashBuffer& manifestSignatureHash,
+        std::vector<ValidationError>& errors)
+    {
+        try
+        {
+            if (!manifestSignatureHash.empty())
+            {
+                auto msixSignature = msixInfo->GetSignature();
```

Comment:
I would consider adding a `GetSignatureHash` method since we are now doing it in two places.  It can even be optimized to not actually store the whole stream in memory, not that it is actually large enough to really matter.